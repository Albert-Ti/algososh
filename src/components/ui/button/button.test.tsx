import { fireEvent, render, screen } from '@testing-library/react'
import { Button } from './button'

describe('Testing the Button component', () => {
  it('Button wit text', () => {
    render(<Button text='Send' />)
    expect(screen.getByRole('button')).toMatchSnapshot()
  })

  it('Button without text', () => {
    render(<Button />)
    expect(screen.getByRole('button')).toMatchSnapshot()
  })

  it('Locked button', () => {
    render(<Button disabled={true} />)
    expect(screen.getByRole('button')).toMatchSnapshot()
  })

  it('button with loading indication', () => {
    render(<Button isLoader={true} />)
    expect(screen.getByRole('button')).toMatchSnapshot()
  })

  it('correct callback call', () => {
    const onClickFake = jest.fn()
    render(<Button onClick={onClickFake} />)

    fireEvent.click(screen.getByRole('button'))
    expect(onClickFake).toHaveBeenCalledTimes(1)
  })
})
