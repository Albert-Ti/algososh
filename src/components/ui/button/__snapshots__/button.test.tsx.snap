// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Button component Button wit text 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    class="text undefined"
  >
    Send
  </p>
</button>
`;

exports[`Testing the Button component Button without text 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;

exports[`Testing the Button component Locked button 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  disabled=""
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;

exports[`Testing the Button component button with loading indication 1`] = `
<button
  class="text text_type_button text_color_primary button undefined loader "
  disabled=""
  type="button"
>
  <img
    alt="Загрузка."
    class="loader_icon"
    src="loader.svg"
  />
</button>
`;
